### PR TITLE
Input fix

### PR DESCRIPTION
**Pull request by merceyz**
An attempt to fix the error with reading the input from binary stream

### CODE REVIEW

File: src/main/predict.cpp [hunk 1]
```diff
@@ -204,8 +204,22 @@ void go(Config config) {
     }
     if(verbose) cout << "inputFile: '" << config.inputFile << "'"<< endl;
     if(config.inputFile == "") {
-        cin.read(reinterpret_cast< char * >(inputData), inputCubeSize * config.batchSize * 4l);
-        more = !cin.eof();
+	const int size = inputCubeSize * config.batchSize * 4l;
```
Comment (hughperkins) [inline src/main/predict.cpp]: here, the size is in bytes, ie multiplied by 4, whereas ...

Comment (hughperkins) [inline src/main/predict.cpp]: oh, I see, you divide by 4 in the other location. ok :-)

File: src/main/predict.cpp [hunk 2]
```diff
@@ -204,8 +204,22 @@ void go(Config config) {
...
+	char *bytes = new char[size];
+
+	cin.read(bytes, size);
+	
+	float *bytes_as_float_pointer = (float*)bytes;
+
+	for (int i = 0; i < size / 4, i++;)
```
Comment (hughperkins) [inline src/main/predict.cpp]: this division will run each iteration of the loop.  its going to be quite fast anyway, but if you move it outside of the loop, then it is only calculated once.

File: src/main/predict.cpp [hunk 3]
```diff
@@ -204,8 +204,22 @@ void go(Config config) {
...
     if(verbose) cout << "inputFile: '" << config.inputFile << "'"<< endl;
     if(config.inputFile == "") {
-        cin.read(reinterpret_cast< char * >(inputData), inputCubeSize * config.batchSize * 4l);
-        more = !cin.eof();
+	const int size = inputCubeSize * config.batchSize * 4l;
+	char *bytes = new char[size];
+
+	cin.read(bytes, size);
+	
+	float *bytes_as_float_pointer = (float*)bytes;
```
Comment (hughperkins) [inline src/main/predict.cpp]: prefer `reinterpret_cast` on the whole.  `(float*)` is from C, I think.

File: src/main/predict.cpp [hunk 4]
```diff
@@ -204,8 +204,22 @@ void go(Config config) {
     }
     if(verbose) cout << "inputFile: '" << config.inputFile << "'"<< endl;
     if(config.inputFile == "") {
-        cin.read(reinterpret_cast< char * >(inputData), inputCubeSize * config.batchSize * 4l);
-        more = !cin.eof();
+	const int size = inputCubeSize * config.batchSize * 4l;
+	char *bytes = new char[size];
+
+	cin.read(bytes, size);
```
Comment (hughperkins) [inline src/main/predict.cpp]: Hmmm, isnt this the same as before?  I thought you were looping over this or something, but it seems like its the exact same commandas before?

**Comment by hughperkins** [comment]
Can you outline your concept of why you feel this is different/better than the original version?  I cant quite see how the change might fix the issue?  Also, I think I'd like to see an explanation of what you feel the issue is.

**Comment by merceyz** [comment]
As i said in the description it was an "attempt". It's not much different but i was hoping it could do the trick as since when we filtered out X bytes to make it work i thought something might be wrong in the reading of the input or converting from the bytes to the float. If not i'm sorry for wasting your time with this

However unable to compile (even though i followed your instructions) i had no way of testing my theory.

Both of them return the same results in my tests however i did not benchmark your code. The for loop is extremely slow at 69093ms when ran 1 million times.
Replacing it with memcpy results in a execution time of 1047ms.

This pull request was an attempt at fixing a problem that i do not know where is nor how to do as i don't know c++.

**Comment by hughperkins** [comment]
Well, thank you for converting my one line of code that runs in ~1second into 10 lines that run in 70 seconds :-P

I think let's continue with diagnosis.  However, I will go from my end.  I will try sending byte values 10, 12, and 13 into deepcl, and see what happens.  If this crashes deepcl, I will fix this.  If deepcl handles them ok, I shall leave you to look into how to send these characters to stdout in c# without causing something strange to happen to them en-route :-)

**Comment by merceyz** [comment]
> Well, thank you for converting my one line of code that runs in ~1second into 10 lines that run in 70 seconds :-P

You're most welcome :laughing:, optimized it also used ~1 second ^^

>  I will try sending byte values 10, 12, and 13

If you look at my comment on the issue you see i said it also crashed when filtering out those

**Pull request closed without merging**